apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -2350,6 +2331,12 @@ where
         ) -> Option<TEdge>,
         override_conditions: &EnabledOverrideConditions,
     ) -> Result<Option<NodeIndex>, FederationError> {
+        // TODO: Temporary fix to avoid optimization if context exists, a permanent fix is here:
+        //       https://github.com/apollographql/federation/pull/3017#pullrequestreview-2083949094
+        if self.graph.is_context_used() {
```

Comment:
Bugs:
- This optimization should be disabled when `@fromContext` is in use, as it is in JS QP.